src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -8359,20 +8555,38 @@ std::string get_parent_dir(const std::string &path) {
 
 Result<ASR::TranslationUnit_t*> python_ast_to_asr(Allocator &al, LocationManager &lm, SymbolTable* symtab,
     AST::ast_t &ast, diag::Diagnostics &diagnostics, CompilerOptions &compiler_options,
-    bool main_module, std::string module_name, std::string file_path, bool allow_implicit_casting)
+    bool main_module, std::string module_name, std::string file_path, bool allow_implicit_casting, bool is_interactive)
 {
     std::map<int, ASR::symbol_t*> ast_overload;
     std::string parent_dir = get_parent_dir(file_path);
-    AST::Module_t *ast_m = AST::down_cast2<AST::Module_t>(&ast);
 
     ASR::asr_t *unit;
-    auto res = symbol_table_visitor(al, lm, symtab, *ast_m, diagnostics, main_module, module_name,
+    AST::Module_t *ast_m = nullptr;
+    AST::Interactive_t *ast_i = nullptr;
+
+    if (!is_interactive) {
+        ast_m = AST::down_cast2<AST::Module_t>(&ast);
+        Result<ASR::asr_t*> res = symbol_table_visitor(al, lm, symtab, *ast_m, diagnostics, main_module, module_name,
         ast_overload, parent_dir, compiler_options.import_paths, allow_implicit_casting);
-    if (res.ok) {
-        unit = res.result;
+        if (res.ok) {
+            unit = res.result;
+        } else {
+            return res.error;
+        }
     } else {
-        return res.error;
+        ast_i = AST::down_cast2<AST::Interactive_t>(&ast);
+        if (!symtab) {
+            // Create new empty symbol table, as a module
+        }
```

Comment:
Do we need this here? Is this a work in progress?

Reply:
This is not required. There are other changes that givens the same effect as creating a `SymbolTable` here.